homeassistant/components/nea/sensor.py

```diff
@@ -0,0 +1,291 @@
+    "psi": ["PSI", None, "mdi:biohazard"],
+    "rainfall": ["Rainfall", LENGTH_MILLIMETERS, "mdi:weather-pouring"],
+    "relative-humidity": ["Humidity", PERCENTAGE, "mdi:water-percent"],
+    "uv-index": ["UV Index", UV_INDEX, "mdi:weather-sunny"],
+    "wind-direction": ["Wind Direction", DEGREE, "mdi:compass-outline"],
+    "wind-speed": ["Wind Speed", SPEED_KILOMETERS_PER_HOUR, "mdi:weather-windy"],
+}
+
+CONDITION_CLASSES = {
+    ATTR_CONDITION_CLEAR_NIGHT: ["Fair (Night)"],
+    ATTR_CONDITION_CLOUDY: ["Cloudy", "Overcast"],
+    ATTR_CONDITION_EXCEPTIONAL: ["Fair (Day)", "Fair & Warm"],
+    ATTR_CONDITION_FOG: ["Mist", "Fog", "Hazy", "Slightly Hazy"],
+    ATTR_CONDITION_HAIL: [],
+    ATTR_CONDITION_LIGHTNING: [],
+    ATTR_CONDITION_LIGHTNING_RAINY: [
+        "Heavy Thundery Showers with Gusty Winds",
+        "Heavy Thundery Showers",
+        "Thundery Showers",
+    ],
+    ATTR_CONDITION_PARTLYCLOUDY: ["Partly Cloudy (Day)", "Partly Cloudy (Night)"],
+    ATTR_CONDITION_POURING: ["Heavy Rain", "Heavy Showers"],
+    ATTR_CONDITION_RAINY: [
+        "Drizzle",
+        "Light Rain",
+        "Light Showers",
+        "Passing Showers",
+        "Moderate Rain",
+        "Showers",
+        "Strong Winds, Showers",
+        "Strong Winds, Rain",
+        "Windy, Rain",
+        "Windy, Showers",
+    ],
+    ATTR_CONDITION_SNOWY: ["Snow"],
+    ATTR_CONDITION_SNOWY_RAINY: ["Snow Showers"],
+    ATTR_CONDITION_SUNNY: ["Sunny"],
+    ATTR_CONDITION_WINDY: ["Strong Winds", "Windy, Cloudy", "Windy", "Windy, Fair"],
+    ATTR_CONDITION_WINDY_VARIANT: [],
+}
+
+CONDITION_ICONS = {
+    ATTR_CONDITION_CLEAR_NIGHT: "mdi:weather-night",
+    ATTR_CONDITION_CLOUDY: "mdi:weather-cloudy",
+    ATTR_CONDITION_EXCEPTIONAL: "mdi:weather-sunny",
+    ATTR_CONDITION_FOG: "mdi:weather-fog",
+    ATTR_CONDITION_HAIL: "mdi:weather-hail",
+    ATTR_CONDITION_LIGHTNING: "mdi:weather-lightning",
+    ATTR_CONDITION_LIGHTNING_RAINY: "mdi:weather-lightning-rainy",
+    ATTR_CONDITION_PARTLYCLOUDY: "mdi:weather-partly-cloudy",
+    ATTR_CONDITION_POURING: "mdi:weather-pouring",
+    ATTR_CONDITION_RAINY: "mdi:weather-rainy",
+    ATTR_CONDITION_SNOWY: "mdi:weather-snowy",
+    ATTR_CONDITION_SNOWY_RAINY: "mdi:weather-snowy-rainy",
+    ATTR_CONDITION_SUNNY: "mdi:weather-sunny",
+    ATTR_CONDITION_WINDY: "mdi:weather-windy",
+    ATTR_CONDITION_WINDY_VARIANT: "mdi:weather-windy-variant",
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_MONITORED_CONDITIONS): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+        vol.Optional(CONF_NAME, default="NEA"): cv.string,
+        vol.Inclusive(
+            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.longitude,
+    }
+)
```

Comment:
We no longer allow integrations to add a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Thanks already! :+1:

Reply:
Hi, thanks for reviewing! Will need some time to refactor it to use config flow. Meanwhile, is there documentation on the proper way to writing tests in HA?

Reply:
I'am subscribed, thank you, I'm working on it

Reply:
Yes, it is on our developer portal, additionally, we have tons of examples in the repo :)